builder/dockerfile/evaluator.go

```diff
@@ -201,3 +201,31 @@ func (b *Builder) dispatch(stepN int, stepTotal int, ast *parser.Node) error {
 
 	return fmt.Errorf("Unknown instruction: %s", upperCasedCmd)
 }
+
+// evaluateDispatch does a simple check for syntax errors of the Dockerfile.
+// Because some of the instructions can only be validated through runtime,
+// arg, env, etc., this syntax check will not be complete and could not replace
+// the runtime check. Instead, this function is only a helper that allows
+// user to find out the obvious error in Dockerfile earlier on.
+func (b *Builder) evaluateDispatch(ast *parser.Node) error {
```

Comment:
IMO this should be called "checkDispatch", since you're not actually evaluating anything here.